### PR TITLE
Cmr 4558

### PR DESCRIPTION
**Pull request by laurenfrederick**
Only show guest permitted collections on the directory page.

Removed the caching for this page and improved performance by hitting elastic directly to get the info instead of getting the umm-json response. This changed the expected response format in data.clj so there are changes to the html and tests to go with that.

### CODE REVIEW

File: common-app-lib/src/cmr/common_app/services/search/params.clj [hunk 1]
```diff
@@ -168,6 +168,16 @@
        :sort-keys (parse-sort-key (:sort-key params) aliases)
        :result-format (:result-format params)}])))
 
+(defn generate-param-query-conditions
+  "Generate the search conditions for the query based on the parameters"
+  [context concept-type params]
+  (let [options (u/map-keys->kebab-case (get params :options {}))
+        params (dissoc params :options)]
+    (when (not (empty? params))
```
Comment (chris-durbin) [inline common-app-lib/src/cmr/common_app/services/search/params.clj]: `(when (seq params))`

**Comment by ygliuvt** [comment]
> Make sure we note for sprint 87 that we need to reindex all of the collections for the new doi-stored field.

Don't we automatically do that every day?

**Comment by chris-durbin** [comment]
>> Make sure we note for sprint 87 that we need to reindex all of the collections for the new doi-stored field.

> Don't we automatically do that every day?
We do once a day, but we want the collections reindexed as soon as possible after deployment to minimize the time where the field is not populated.

File: search-app/src/cmr/search/site/data.clj [hunk 2]
```diff
@@ -66,18 +69,33 @@
...
+                                  :result-format :query-specified
+                                  :result-fields [:concept-id
+                                                  :doi-stored
+                                                  :entry-title
+                                                  :short-name
+                                                  :version-id]})
+        result (query-exec/execute-query context query)]
+    (sort-by :entry-title (:items result))))
+
+(defn-timed collection-data :default
```
Comment (ygliuvt) [inline search-app/src/cmr/search/site/data.clj]: Isn't this a syntax error? why change `defmethod` to `defn-timed`?

**Comment by ygliuvt** [comment]
>>>Make sure we note for sprint 87 that we need to reindex all of the collections for the new doi-stored field.
>>Don't we automatically do that every day?
>We do once a day, but we want the collections reindexed as soon as possible after deployment to minimize the time where the field is not populated.
Unless we backport this fix right now, otherwise it is an indication that it is not worth the effort. I have not thought this is a hot issue.

**Comment by chris-durbin** [comment]
> Unless we backport this fix right now, otherwise it is an indication that it is not worth the effort. I have not thought this is a hot issue.

I'm not sure what you are saying about backporting, but hopefully this helps clear up what I'm saying. Our process for new indexed fields for granules and collections is different. Any granule index changes we make over the course of two sprints. One sprint to start indexing the new field and allowing bulk indexing to get everything and then the next sprint we can update search to use the new field.

We decided a long time ago that for collection updates we would allow the index change and the search change to happen together. The idea was we could index the collections fast enough that within an hour of deployment everything would work as expected, so it wasn't worth the overhead of making the changes across two different sprints. Since no exceptions are thrown, its just that searches using the parameter won't find all the correct results until indexing completes, we all agree it's ok to proceed this way. We just make sure to index the collections as soon after deployment as possible.

File: search-app/src/cmr/search/site/data.clj (hunk 2 above)
Comment (laurenfrederick) [reply search-app/src/cmr/search/site/data.clj]: Yes I had it in a separate method and must have done an undo accidentally or something right before committing. I'm going to fix now.

**Comment by ygliuvt** [comment]
What I am saying is that although there is nothing wrong to manually reindex all collections so that the newly added elasticsearch field becomes searchable right after new deployment. We can relax the manual reindexing process as the reindex will be automatically done at 12:20pm, which is probably right after the deployment, and we don't really care if the auto reindex takes a little longer to kick in as the fact that we don't backport this change immediately is an indication that we don't care if this new field becomes searchable asap. It's not important. it's just that I am for less manual intervention.

File: search-app/src/cmr/search/site/data.clj (hunk 2 above)
Comment (laurenfrederick) [reply search-app/src/cmr/search/site/data.clj]: Looks like I hadn't saved what I was doing. Fixed. Thanks!!

File: common-app-lib/src/cmr/common_app/services/search/params.clj (hunk 1 above)
Comment (laurenfrederick) [reply common-app-lib/src/cmr/common_app/services/search/params.clj]: Fixed, thanks!

**Pull request merged**